### PR TITLE
[#713] app boilerplate to files

### PR DESCRIPTION
**Pull request by voluntatis-summ**
Closes: #713

## Description

Make app.go boilerplate methods go into their own file
- added an appKeepers struct for the OsmosisApp to inherit listed keepers
- moved several modules list to app/modules.go

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by ValarDragon** [comment]
This is great! Think we can make AppKeepers its own module, and the AppKeepers struct publicly exposed as well?

**Comment by voluntatis-summ** [comment]
What do you mean exactly when you say separate module?

**Comment by ValarDragon** [comment]
a separate folder / go package

**Comment by voluntatis-summ** [comment]
https://github.com/osmosis-labs/osmosis/blob/d84ba8db98ba9d5392db0b1c41bf498b6999112a/app/keepers.go#L174-L179

https://github.com/osmosis-labs/osmosis/blob/d84ba8db98ba9d5392db0b1c41bf498b6999112a/app/keepers.go#L196-L198

https://github.com/osmosis-labs/osmosis/blob/d84ba8db98ba9d5392db0b1c41bf498b6999112a/app/keepers.go#L218-L220

few dependencies between keepers logic and OsmosisApp struct

What would be the cleanest of separation of concerns here? 

I think it would be to pass in OsmosisApp struct to AppKeepers functions(InitSpecialKeepers, InitNormalKeepers), but it would cause a circular dependency atm with app.go < - > AppKeepers because AppKeepers need would need struct definition and app.go uses the AppKeepers methods

**Comment by ValarDragon** [comment]
Lets pass everything it needs from app as a parameter to the functions. We can separately file issues for pairing down the number of args in the future.

A separate appkeepers struct makes the upgrade architecture much more tenable as well.

**Comment by voluntatis-summ** [comment]
moving AppKeepers to a separate module not working as cleanly as I hoped... having issues with the pointers and uninitialized keepers

perhaps we split off the "moving AppKeepers to a new package" into a new issue?
I could pick it up again from there?

**Comment by ValarDragon** [comment]
Sure!

**Comment by ValarDragon** [comment]
Thanks for the work on this! Unfortunately, cannot get this reviewed before the v7 code freeze. I definitely think this is awesome, and want to get it into a v7.1

File: app/keepers.go [hunk 1]
```diff
@@ -380,3 +478,26 @@ func (app *OsmosisApp) SetupHooks() {
 		),
 	)
 }
+
+// initParamsKeeper init params keeper and its subspaces
+func (app *OsmosisApp) initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino, key, tkey sdk.StoreKey) paramskeeper.Keeper {
```
Comment (ValarDragon) [inline app/keepers.go:483]: Should we remove the `(app *OsmosisApp)` part?

File: app/modules.go [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+		ibctransfertypes.ModuleName,
+		authtypes.ModuleName,
+		banktypes.ModuleName,
+		govtypes.ModuleName,
+		crisistypes.ModuleName,
+		genutiltypes.ModuleName,
+		authz.ModuleName,
+		paramstypes.ModuleName, vestingtypes.ModuleName,
+		gammtypes.ModuleName, incentivestypes.ModuleName, lockuptypes.ModuleName, claimtypes.ModuleName,
+		poolincentivestypes.ModuleName, superfluidtypes.ModuleName, bech32ibctypes.ModuleName, txfeestypes.ModuleName,
```
Comment (ValarDragon) [inline app/modules.go:255]:
```suggestion
		poolincentivestypes.ModuleName, 
		// superfluid must come after distribution and epochs
		superfluidtypes.ModuleName, 
		bech32ibctypes.ModuleName, txfeestypes.ModuleName,
```

**Review by ValarDragon — APPROVED** [review]
Awesome work on this! TYSM

Reviewed it line by line, looks great to me!

**Pull request merged**